homeassistant/components/homekit_controller/climate.py

```diff
@@ -57,10 +57,26 @@ def get_characteristic_types(self):
     def _setup_heating_cooling_target(self, characteristic):
         self._features |= SUPPORT_OPERATION_MODE
 
-        valid_values = characteristic.get(
-            'valid-values', DEFAULT_VALID_MODES)
+        if 'valid-values' in characteristic:
+            valid_values = filter(
+                lambda val: val in DEFAULT_VALID_MODES,
```

Comment:
Use a list comprehension instead of filter + lambda.

Reply:
Done this one. Do you want all three doing?  (I have quickly tried it and they look ugly but will do as i'm told :-D)

Reply:
Yes. We should basically never use filter + lambda. Comprehensions are easier to read.

Reply:
Fixed them all and rebased to pick up the CI config change.